### PR TITLE
Upgraded GDAL to version 2.3.0.

### PR DESCRIPTION
**Pull request by chetnieter**
Upgraded GDAL from version 2.2.2 to 2.2.4 and made 2.2.4 the default
version.

### CODE REVIEW

**Comment by chetnieter** [comment]
@dstoup PTAL

**Comment by dstoup** [comment]
jenkins test this please

**Comment by dstoup** [comment]
One comment about the patch, please update release notes. Otherwise, it looks good pending dashboard builds

**Comment by dstoup** [comment]
Jenkins test this please

**Comment by chetnieter** [comment]
So it turns out I needed 2.3.0 not 2.2.3 for the new API. I switched everything to 2.3.0.

**Comment by chetnieter** [comment]
Jenkins test this please

File: Doc/release-notes/master.txt [hunk 1]
```diff
@@ -92,6 +92,9 @@ Package Upgrades
 
  * OpenCV 3.4.0 is now the default.
 
+ * Update GDAL to 2.2.4 which is now the default. Maintaining the option
```
Comment (dstoup) [inline Doc/release-notes/master.txt]: This comment needs to be updated to reflect the new version change.

**Comment by dstoup** [comment]
Jenkins test this please

**Pull request merged**